src/main/java/com/toofifty/easygiantsfoundry/EasyGiantsFoundryPlugin.java

```diff
@@ -161,6 +197,7 @@ public void onItemContainerChanged(ItemContainerChanged event)
 		 	&& event.getItemContainer().count(PREFORM) == 0)
 		{
 			state.reset();
+			oldStage = null;
```

Comment:
oldStage needs to also be set to null when you either hand in or start a new commision, the objects only despawn when you leave the minigame.

Reply:
> Isn't this triggered when you hand in or start a new commision?

You're correct, my mistake. Github shortened the code and it looked like it was inside the onGameObjectDespawned method.